Make export._trace._WrapperModule work in strict mode

yushangdi opened the pull request:
Summary:
as title

`export._trace._WrapperModule` is used to wrap functions into a Module so we can export the function.

We add `export._wrapper_utils` to `dynamo`'s `MOD_INLINELIST` so dynamo traces into `_WrapperModule`


Fixes https://github.com/pytorch/pytorch/issues/146867

Test Plan:
```
 buck run fbcode//mode/dev-nosan //caffe2/test:test_export -- -r wrapper_module
```

Differential Revision: D69434316




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

angelayi commented on test/export/test_export.py:
```diff
@@ -11881,6 +11881,30 @@ def forward(self, x):
+
+        from torch.export import _trace
+
+        model = _trace._WrapperModule(f)
```
I would prefer to move this to some utils file, instead of skipfiling the entire trace.py

henrylhtsang commented:
Noob question, is there any chance we can make it work for module method as well?


cc @henryhu6 @chenyang78 looks like it can work for functions

yushangdi commented:
> Noob question, is there any chance we can make it work for module method as well?
> 
> cc @henryhu6 @chenyang78 looks like it can work for functions

@henrylhtsang I think it works for module method. Do you mean this? The code below runs fine for me:

```
import torch
from torch.export import _trace
            
class ToTuple(torch.nn.Module):
    def test(self, x):
        return (torch.abs(x),)

model_inner = ToTuple()
model = _trace._WrapperModule(model_inner.test)
ep = torch.export.export(model, (torch.randn(8,),), strict=False)
```

henrylhtsang commented:
Can we consider landing this? 

My use case uses strict=False, so this is not urgent, but would be nice if both modes are supported.

yushangdi committed:
Make export._trace._WrapperModule work in strict mode (#146919)

Summary:

as title

`export._trace._WrapperModule` is used to wrap functions into a Module so we can export the function.

We add `export._trace` to `dynamo`'s `MOD_INLINELIST` so dynamo traces into `_WrapperModule`


Fixes https://github.com/pytorch/pytorch/issues/146867

Test Plan:
```
buck run fbcode//mode/dev-nosan //caffe2/test:test_export -- -r wrapper_module
```

Differential Revision: D69434316

yushangdi commented:
> Can we consider landing this?
> 
> My use case uses strict=False, so this is not urgent, but would be nice if both modes are supported.

@henrylhtsang I just updated the Diff. After all CI pass, I'll land it.

clee2000 commented:
@pytorchbot merge -f "merged internally"

malfet commented:
@pytorchbot revert -m "Broke lint, see https://github.com/pytorch/pytorch/actions/runs/14415686353/job/40431799827" -c nosignal
